homeassistant/components/google_assistant/helpers.py

```diff
@@ -327,7 +332,25 @@ def __init__(self, hass):
         """Initialize a configuration store."""
         self._hass = hass
         self._store = Store(hass, self._STORAGE_VERSION, self._STORAGE_KEY)
-        self._data = {STORE_AGENT_USER_IDS: {}}
+        self._data = None
+
+    async def async_initialize(self):
+        """Finish initializing the ConfigStore."""
+        if (data := await self._store.async_load()) is None:
+            data = {
+                STORE_AGENT_USER_IDS: {},
+            }
+            await self._store.async_save(data)
```

Comment:
you're saving twice here. If you use delay_save you will only write once.

Reply:
or track in a boolean if you need to write.

Reply:
Used the boolean to track.